frontend/src/app/modules/main/cards/constructor-container-card/constructor-container-card/constructor-container-card.component.ts

```diff
@@ -0,0 +1,48 @@
+import {
+  Component, EventEmitter, Input, Output
+} from '@angular/core';
+import { Router } from '@angular/router';
+import { ContainerView } from 'src/app/models/constructor/container-view';
+import { ConstructorService } from 'src/app/services/constructor.service';
+import { SnackbarService } from 'src/app/services/snackbar.service';
+
+@Component({
+  selector: 'app-constructor-container-card',
+  templateUrl: './constructor-container-card.component.html',
+  styleUrls: ['./constructor-container-card.component.sass']
+})
+export class ConstructorContainerCardComponent {
+  @Input()
+  container: ContainerView = {} as ContainerView;
+
+  @Output()
+  clickEmiter = new EventEmitter<void>();
+
+  constructor(
+    private _constructorService: ConstructorService,
+    private _router: Router,
+    private _snackbarService: SnackbarService
+  ) { }
+
+  publishContainer = () => {
+    this._constructorService.publishContainer(this.container).subscribe();
```

Comment:
There is a subscription without unsubscribing. Use takeUntil, take and so on `this._constructorService.publishContainer(this.container).subscribe()`